scripts/Modules/Saritasa.Cake/Tasks/PackTask.cs

```diff
@@ -0,0 +1,164 @@
+    public const string AssemblyFileVersionAttribute = "AssemblyFileVersion";
+    public const string AssemblyInformationalVersionAttribute = "AssemblyInformationalVersion";
+
+    /// <summary>
+    /// Constructor.
+    /// </summary>
+    public PackTask()
+    {
+        projectNamesForPack = new[]
+        {
+            "Saritasa.Tools.Common",
+            "Saritasa.Tools.Domain",
+            "Saritasa.Tools.EntityFrameworkCore",
+            "Saritasa.Tools.Emails",
+            "Saritasa.Tools.Misc",
+        };
+    }
+
+    private string[] projectNamesForPack;
+
+    /// <inheritdoc />
+    public override void Run(PackContext context)
+    {
+        UpdateAssemlyVersions(context);
+
+        PackProjects(context);
+
+        RevertAssemlyVersions(context);
+    }
+
+    private void UpdateAssemlyVersions(PackContext context)
+    {
+        var revcount = context.ExecuteGitCommand("rev-list --all --count").Replace(Environment.NewLine, string.Empty);
+        var hash = context.ExecuteGitCommand("log --pretty=format:%h -n 1").Replace(Environment.NewLine, string.Empty);
+
+        foreach (var projectName in projectNamesForPack)
+        {
+            var assemblyVersion = GetProjectAssemblyVersion(context, projectName).ToString();
+            var fileVersion = $"{GetVersion(context, projectName)}.{revcount}";
+            var productVersion = $"{fileVersion}-{hash}";
+            context.Information($"{projectName} has versions {assemblyVersion} {fileVersion} {productVersion}");
+
+            ReplaceAttributeValueInAssemblyInfo(context, projectName, AssemblyVersionAttribute, assemblyVersion);
+            ReplaceAttributeValueInAssemblyInfo(context, projectName, AssemblyFileVersionAttribute, fileVersion);
+            ReplaceAttributeValueInAssemblyInfo(context, projectName, AssemblyInformationalVersionAttribute, productVersion);
+        }
+    }
+
+    private void PackProjects(PackContext context)
+    {
+        foreach (var projectName in projectNamesForPack)
+        {
+            var projectDir = $"{context.SolutionDir}src\\{projectName}";
+
+            context.DotNetRestore(projectDir);
+            context.DotNetBuild(projectDir, new DotNetBuildSettings
+            {
+                Configuration = context.MsBuildConfiguration
+            });
+
+            PackNuGetFile(context, projectName, projectDir);
+        }
+    }
+
+    private void PackNuGetFile(PackContext context, string projectName, string projectDir)
+    {
+        var longHash = context.ExecuteGitCommand("rev-parse HEAD").Replace(Environment.NewLine, string.Empty);
+        var nuspecFile = $"{projectDir}\\{projectName}.nuspec";
+
+        context.CopyFile($"{nuspecFile}.template", nuspecFile);
+
+        UpdateVariableInFile(nuspecFile, "CommitHash", longHash);
+
+        var assemblyVersion = GetVersion(context, projectName);
+        var resultPackNuGet = context.ExecuteNuGetCommand($"pack {nuspecFile} " +
```

Comment:
Can we use extension for that? https://cakebuild.net/api/Cake.Common.Tools.NuGet/NuGetAliases/EF4ED944